src/data_utils.py

```diff
@@ -0,0 +1,107 @@
+import os
+from dataclasses import dataclass
+
+from wav2vec2 import Wav2Vec2Processor
+import tensorflow as tf
+import tensorflow_io as tfio
+
+
+@dataclass
+class DataLoader:
+    data_dir: str = "../data/LibriSpeech/data"
+    batch_size: int = 16
+    audio_maxlen: int = 400000
+    pad_id: int = 0
+
+    processor: Wav2Vec2Processor = Wav2Vec2Processor(is_tokenizer=False)
+
+    dataset: str = "librispeech" # timit
+    buffer_size: int = 10000
```

Comment:
Since this class is decorated with dataclass, they are not static (you might have missed this [line](https://github.com/vasudevgupta7/gsoc-wav2vec2/blob/7431f3abf25e260d3b129af08b8ec3b5146f3bb8/src/data_utils.py#L9) I think). Are you suggesting to switch to normal class from dataclass??

Reply:
I would prefer to have these wrapped under `__init__()`. Are there advantages of using the decorator?

Reply:
with __init__(), we will have to write same lines multiple times. and we get __repr__ automatically.

Example:

```python
# without dataclass
class hello:
   def __init__(self, a):
       self.a = a
   def __repr__(self):
       return f"{self.__class__.__name__}({a=<some value>})"

# but with dataclass:
@dataclass
class hello:
   a: int
```

Google's flax is also built on dataclass like structure, So I prefer using them. Please confirm if it's fine to use dataclasses here.

Reply:
While the Google style-guide does support the use of dataclasses, it is really meant for struct-like classes that just contain some data (eg. a Point class with X and Y values). However, this looks like more of a functional class, so I don't think the dataclass annotation is appropriate.

Reply:
Okay, I will shift to normal classes. Thanks!!

Reply:
From Sayak:

> Also, have you hosted the converted checkpoints somewhere? It's fine I think. We could also probably have it uploaded to TensorFlow Hub as we are working. @MorganR to comment on this more. When you are downloading the checkpoints and loading them inside the model class, a descriptive message would be better for the end-user.

Sounds like these are hosted on Hugging Face during development. That sounds reasonable. I agree with your plan to wait until these are polished and fine-tunable before publishing them to Hub.